### PR TITLE
Spacepod Cargo Bay Equipment and Pod Fabricator Spruce Up

### PR DESCRIPTION
**Pull request by jwhitak**
# Two in one, more like unatomic

POV: you're a crate getting picked up by a space pod

[cratelife.webm](https://github.com/user-attachments/assets/faf7abff-db28-48bd-b73a-1c32bd0f17e2)

## What this does
This adds a new piece of spacepod equipment called the crate cargo bay. It allows a space pod to store a single crate. 
image

This can be useful in many ways: allowing a vault hunter to carry more loot, giving a paramedic a place to stuff cadavers, and even safely delivering a supermatter crate to just outside the escape shuttle! It requires a small amount of materials research and a good amount of iron, and can be printed at the pod fabricator.

You can load the crate either outside or inside the pod, but it takes 4 seconds to load or unload, and the pod must remain still.

The pod's internal tank atmosphere is used for anything that happens to be stuffed in the crate. Be warned!

And in the ultimate unatomic PR, I touched up the pod fabricator while I was here because the UI was terrible.
image
Now it's less terrible, mimicing the exosuit fabricator. Features a Quick Pod button to auto-queue the basic pod frame, removed useless "parts" category, and always displays its queue.

Also fixed a runtime that happened when space pods exploded.

## Why it's good
Get like 10 dudes resting in a crate, load it up, and then try to get to places that you probably shouldn't.

## How it was tested
Many ways, but this one where I loaded a frozen cadaver in a freezer crate into a space pod and blew it up was particularly interesting.
image
image
image

## Changelog

:cl:
 * rscadd: Added spacepod cargo bays, where you can upgrade your pod to hold a single crate.
 * tweak: Pod Fabricator is less terrible.

### CODE REVIEW

**Comment by SECBATON-GRIFFON** [comment]
makes passenger seats useless aside from personal preference of riding but regardless, real fun PR

**Comment by SonixApache** [comment]
So what happens if you're a guy hiding in a crate that gets loaded in the pod, are you just stuck in there forever?

**Comment by jwhitak** [comment]
Yes.

**Comment by SonixApache** [comment]
out of PR scope, but something should probably be done about that
does it only take crates or does it work with closets as well?

**Comment by jwhitak** [comment]
Only crates with this particular piece of equipment.

**Comment by jwhitak** [comment]
DNM until I add a means to escape permanent pod entrapment, shouldn't take more than a day or so before I can get to it.

**Comment by jwhitak** [comment]
Ok, you can escape now. Takes 30 seconds after resisting.

**Comment by SECBATON-GRIFFON** [comment]
> Ok, you can escape now. Takes 30 seconds after resisting.

a minute makes a bit more sense, it should be whatever the length of resisting out of a welded locker is

**Comment by jwhitak** [comment]
It only ejects the crate out of the pod, it doesn't unlock or otherwise resist out of the crate.

Also the pod must not be moving (do_after...).

File: code/modules/mob/living/living.dm [hunk 1]
```diff
@@ -1022,6 +1022,11 @@ Thanks.
 			var/obj/structure/rack/crate_shelf/CS = C.loc
 			CS.relay_container_resist_act(src,R)
 			return
+		if(istype(C.loc, /obj/spacepod/) && istype(C,/obj/structure/closet/crate)) //todo - make this generic for future space pod cargo systems
```
Comment (west3436) [inline code/modules/mob/living/living.dm:1025]: Make an issue report for this todo so we don't lose it

File: code/modules/spacepods/spacepods.dm [hunk 2]
```diff
@@ -92,6 +92,8 @@
 		for(var/mob/living/L in occupants)
 			move_outside(L)
 			L.gib()
+	if(ES && ES.cargo_system)
```
Comment (west3436) [inline code/modules/spacepods/spacepods.dm:95]:
```suggestion
	if(ES?.cargo_system)
```

File: code/modules/spacepods/spacepods.dm [hunk 3]
```diff
@@ -140,10 +142,13 @@
 		spawn(0)
 			var/mob/living/L = get_pilot()
 			if(L)
-
 				to_chat(L, "<big><span class='warning'>Critical damage to the vessel detected, core explosion imminent!</span></big>")
-			for(var/i = 10, i >= 0; --i)
+			if(ES && ES.cargo_system && ES.cargo_system.stored)
```
Comment (west3436) [inline code/modules/spacepods/spacepods.dm:146]:
```suggestion
			if(ES?.cargo_system?.stored)
```

File: code/modules/spacepods/spacepods.dm [hunk 4]
```diff
@@ -182,6 +190,8 @@
 				if(istype(AM,/obj/item))
 					if(AM == battery || istype(AM, /obj/item/device/spacepod_equipment))
 						continue //don't eject this particular item!
+					if(ES && ES.cargo_system && istype(AM, ES.cargo_system.allowed_types))
```
Comment (west3436) [inline code/modules/spacepods/spacepods.dm:193]:
```suggestion
					if(ES?.cargo_system && istype(AM, ES.cargo_system.allowed_types))
```

File: code/modules/spacepods/spacepods.dm [hunk 5]
```diff
@@ -393,12 +425,14 @@
 			. = t_air.return_temperature()
 	return
 
-/obj/spacepod/MouseDropTo(mob/M, mob/user)
-	if(M != user)
+/obj/spacepod/MouseDropTo(atom/moved, mob/user)
+	if(!Adjacent(moved) || !Adjacent(user))
 		return
-	if(!Adjacent(M) || !Adjacent(user))
+	if(ES && ES.cargo_system && is_type_in_list(moved, ES.cargo_system.allowed_types))
```
Comment (west3436) [inline code/modules/spacepods/spacepods.dm:431]:
```suggestion
	if(ES?.cargo_system && is_type_in_list(moved, ES.cargo_system.allowed_types))
```

File: code/modules/spacepods/spacepods.dm [hunk 6]
```diff
@@ -463,6 +497,94 @@
 		to_chat(usr, "You stop entering the pod.")
 	return
 
+/obj/spacepod/proc/attempt_load_cargo(atom/movable/moved, mob/user)
+	if(!ES || !istype(ES))
+		to_chat(user, "<span class='warning'>The pod has no equipment datum, or is the wrong type, yell at pomf.</span>")
```
Comment (west3436) [inline code/modules/spacepods/spacepods.dm:502]:
```suggestion
		to_chat(user, "<span class='warning'>The pod has no equipment datum or is the wrong type - please bugreport this.</span>")
		CRASH("[user] attempt_load_cargo() was called despite [src] lacking a valid equipment datum.")
```

File: code/modules/spacepods/spacepods.dm [hunk 7]
```diff
@@ -463,6 +497,94 @@
...
+	if(locked)
+		to_chat(usr, "<span class = 'warning'>\The [src] is locked.</span>")
+		return
+	if(usr.incapacitated() || usr.lying) //are you cuffed, dying, lying, stunned or other
+		return
+	if (!ishigherbeing(usr))
+		return
+	if(!ES.cargo_system.stored)
+		to_chat(usr, "<span class='warning'>The pod has nothing in the cargo bay.</span>")
+		return
```
Comment (west3436) [inline code/modules/spacepods/spacepods.dm:553]: You've used this block twice; time to turn it into a proc :^)
Same comment for line 539 applies as in the last instance of this

**Review by west3436 — APPROVED** [review]

**Pull request merged**